tests/cases/fourslash/getOutliningSpansForRegions.ts

```diff
@@ -0,0 +1,46 @@
+/// <reference path="fourslash.ts"/>
+
+////// basic region
+////[|// #region
+////
+////// #endregion|]
+////
+////// region with label
+////[|// #region label1
+////
+////// #endregion|]
+////
+////// region with extra whitespace in all valid locations
+////[|             //              #region          label2    label3
+////
+////        //        #endregion|]
+////
+////// No space before directive
+////[|//#region label4
+////
+//////#endregion|]
+////
+////// Nested regions
+////[|// #region outer
+////
+////[|// #region inner
+////
+////// #endregion inner|]
+////
+////// #endregion outer|]
+////
+////// region delimiters not valid when preceding text on line
+//// test // #region invalid1
+////
+////test // #endregion
+////
+////// region delimiters not valid when in multiline comment
+/////*
+////// #region invalid2
+////*/
+////
+/////*
+////// #endregion

```

Comment:
It would be good to have tests with unbalanced start/end markers too. This is an area where a regression could easily slip through and cause exceptions.